### PR TITLE
[GH-936] Fix extended_chip_manifest file

### PR DESCRIPTION
**Pull request by samanehsan**
### Purpose
Fixes: https://broadinstitute.atlassian.net/browse/GH-936

### Changes
Construct the extended_chip_manifest_file from the `cloudChipMetaDataDirectory` and the `extendedIlluminaManifestFileName`.

### Review Instructions

- No instructions.

### CODE REVIEW

File: src/ptc/util/misc.clj [hunk 1]
```diff
@@ -125,3 +125,14 @@
...
     value))
+
+(defn parse-gs-url
+  "Return BUCKET and OBJECT from a gs://bucket/object URL."
+  [url]
+  (let [[gs-colon nada bucket object] (str/split url #"/" 4)]
+    (when-not
+      (and (every? seq [gs-colon bucket])
+           (= "gs:" gs-colon)
+           (= "" nada))
```
Comment (jack-r-warren) [inline src/ptc/util/misc.clj]:
```suggestion
  (let [[_ _ bucket object] (str/split url #"/" 4)]
    (when-not
      (and (seq bucket) (str/starts-with? url "gs://"))
```
I don't know if this is much cleaner, sorta just practicing my Clojure and thought I'd suggest it

Comment (jack-r-warren) [reply src/ptc/util/misc.clj]: Looks like my suggestion got messed up, so nevermind 🙂

File: src/ptc/util/jms.clj [hunk 2]
```diff
@@ -140,13 +139,26 @@
       (apply misc/shell! "gsutil" "cp" (concat sources [cloud]))
       (reduce cloudify (reduce rekey {} copy) chip-and-push))))
 
+(defn get-extended-chip-manifest
+  "Get the extended_chip_manifest_file from the JMS message"
+  [workflow]
+  (let [aou-reference-bucket (misc/getenv-or-throw "AOU_REFERENCE_BUCKET")
+        cloud-chip-metadata-dir (get workflow :cloudChipMetaDataDirectory)
```
[the remark this answers is not in the document]
Comment (samanehsan) [reply src/ptc/util/jms.clj:148]: Good point! I updated the list of required fields

File: src/ptc/util/jms.clj [hunk 3]
```diff
@@ -140,13 +141,26 @@
       (apply misc/shell! "gsutil" "cp" (concat sources [cloud]))
       (reduce cloudify (reduce rekey {} copy) chip-and-push))))
 
+(defn get-extended-chip-manifest
+  "Get the extended_chip_manifest_file from the JMS message"
+  [workflow]
+  (let [aou-reference-bucket (or (System/getenv "AOU_REFERENCE_BUCKET") "broad-arrays-dev-storage")
+        cloud-chip-metadata-dir (get workflow :cloudChipMetaDataDirectory)
+        bucket (first (misc/parse-gs-url cloud-chip-metadata-dir))
+        aou-chip-metadata-dir (str/replace cloud-chip-metadata-dir bucket aou-reference-bucket)
```
Comment (rexwangcc) [inline src/ptc/util/jms.clj:150]: I thought we need `(re-pattern bucket)` 🤔

Comment (samanehsan) [reply src/ptc/util/jms.clj:150]: It can take either a string or a pattern https://clojuredocs.org/clojure.core/replace

**Pull request merged**